README.md

```diff
@@ -6,3 +6,112 @@ AiiDA plugin that simplifies working with pseudo potentials.
+
+If you use `aiida-quantumespresso` the `pseudos` dictionary returned by `get_pseudos` can be directly used as an input for `PwCalculation`s.
+
+## Design
+
+The plugin is centered around two concepts: pseudopotentials and pseudopotential families.
+The two concepts are further explained below, especially focusing on how they are implemented in `aiida-pseudo`, what assumptions are made and what the limitations are.
+
+### Pseudopotentials
+
+Pseudopotentials are implemented as [data plugins](https://aiida-core.readthedocs.io/en/latest/topics/data_types.html#creating-a-data-plugin) and the base class is `PseudoPotentialData`.
+Pseudopotentials are assumed to be defined by a single file on disk and represent a single chemical element.
+As such, each pseudopotential node, _has_ to have two attributes: the md5 of the file that it represents and the symbol of chemical element that the pseudopotential describes.
+The latter follows IUPAC naming conventions as used in the module `aiida.common.constants.elements` of `aiida-core`.
+
+The `PseudoPotentialData` functions as a base class and does not represent an actual existing pseudopotential format.
+It _is_ possible to create a family of `PseudoPotentialData` nodes, but since the element cannot be parsed from the file content itself, it will have to be done from the _filename_, so the filename _has_ to have the format `CHEMICAL_SYMBOL.EXTENSION`, for example `Ar.pseudo`.
+Since the file does not represent an actual pseudopotential, this class is mostly useful for development.
+
+The plugin comes with a variety of pseudopotential subtypes that represent various common pseudopotential formats, such as UPF, PSF, PSML and PSP8.
+The corresponding data plugins implement how the element is parsed from a pseudopotential file with such a format.
+A new pseudopotential node can be created by instantiating the corresponding plugin class:
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+
+    with open('path/to/pseudo.upf', 'rb') as stream:
+        pseudo = UpfData(stream)
+
+Note, the pseudopotential constructor expects a stream of bytes, so be sure to open the file handle with the `'rb'` mode.
+Alternatively, you can also use the `get_or_create` classmethod, which will first search the database if a pseudopotential of the exact same type and content already exists.
+If that is the case, that node is returned, otherwise a new instance is created:
+
+    from aiida import plugins
+
+    UpfData = plugins.DataFactory('pseudo.upf')
+
+    with open('path/to/pseudo.upf', 'rb') as stream:
+        pseudo = UpfData.get_or_create(stream)
+
+If a new node was created, it will be unstored.
+Note that if more than one pseudopotentials in the database are matched, a random one is selected and returned.
+
+
+### Families
+
+The pseudopotentials are organized in families, which are implemented as group plugins and the base class is `PseudoPotentialFamily`.
```

Comment:
Having loose pseudopotentials floating around is not very practical, so groups of related pseudopotentials can be organized into "families", which are implemented as group plugins with the base class `PseudoPotentialFamily`.